react-js/react-components-and-hierarchies.md

```diff
@@ -0,0 +1,438 @@
+      <Button color="warning">warning</Button>
+      <Button color="danger">danger</Button>
+      <Button color="link">link</Button>
+    </div>
+  );
+   }
+
+}
+
+export default App;
+```
+[Run code](https://codepen.io/annu12340/pen/QWjpMRK?editors=1010)
+
+
+#### Reactstrap Cards
+
+```
+
+
+class ReactstrapCard extends React.Component{
+  
+  render(){
+    return(
+      <div className = "container mx-auto">
+
+        <div className = "row">
+          <badge className="badge badge-pill badge-success mx-auto text-center">Active</badge>
+        </div>
+        <div className = "row">
+          <h1 className="text-muted text-center mx-auto">Casey Cupcake</h1>
+          </div>
+        <div className = "row">
+          <h4 className="text-center mx-auto">Graphic Cupcake Designer</h4>
+          </div>
+        <div className = "row">
+          <email className="mx-auto">cupcake@cupcakse.org</email>
+        </div>
+        <div className = "row">
+          <button type="button" className="btn btn-primary mx-auto">
+             Messages <span className="badge badge-light">4</span>
+         </button>
+        </div>
+      </div>
+      );
+  }
+}
+
+ReactDOM.render(<ReactstrapCard/>,document.getElementById('app'));
+
+```
+[Run code](https://codepen.io/CandiceL/pen/MQwyYp?editors=0010)
+
+#### Implementing Reactstrap Modal in React
+The reactstrap Modal component creates a Bootstrap Modal with a header, a body, and a footer.
+```
+<Modal isOpen={this.state.show} toggle={this.toggleModal} >
+
+    <ModalHeader toggle={this.toggle}>
+        Modal title
+    </ModalHeader>
+
+    <ModalBody>
+        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
+    </ModalBody>
+
+    <ModalFooter>
+        <Button color="primary" onClick={this.toggle}>Do Something</Button>{' '}
+        <Button color="secondary" onClick={this.toggle}>Cancel</Button>
+     </ModalFooter>
+    </Modal>
+```
+[Run code](https://codepen.io/annu12340/pen/dyYvVWB?editors=1010)
+
+### Guided Practice
+> #### Let's create a random Quote Generator using react. The application would display a random text from an array when a user clicks on 'new quote' button
+[View demo](https://u79vw.csb.app/) <br/>
+[View code](https://codesandbox.io/s/random-quote-machine-u79vw?file=/src/components/QuoteText.js)
+- First, we need to create a React application.
+	- Create a React app using create React app command. `npx create-react-app`.
+    - This will create a simple React app , and now we will start making changes to this app.
+	- First of all get rid of all the unnecessary stuff showing on DOM , remove all the lines in render of `App.js`. 
+
+ #### Create a json file with some quotes
+ First let us create an array of random quotes and save it in a json file
+ ```
```

Comment:
Can this just be a `const` in the file? Why use JSON? The goal is to keep the code as simple as possible to teach the concept.